internal/services/phone/callhandling/call_handling_business_hours_resource.go

```diff
@@ -0,0 +1,900 @@
+	PhoneNumber types.String `tfsdk:"phone_number"`
+}
+
+func (r *tfBusinessHoursResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state businessHoursResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	output, err := r.read(ctx, &state)
+	if err != nil {
+		resp.Diagnostics.AddError("Error reading phone call handling", err.Error())
+		return
+	}
+
+	diags = resp.State.Set(ctx, &output)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *tfBusinessHoursResource) read(ctx context.Context, plan *businessHoursResourceModel) (*businessHoursResourceModel, error) {
+	dto, err := r.crud.readBusinessHours(ctx, plan.ExtensionID)
+	if err != nil {
+		return nil, fmt.Errorf("error read: %v", err)
+	}
+	if dto == nil {
+		return nil, nil // already deleted
+	}
+
+	customHoursSetting := lo.Ternary(plan.CustomHours != nil && plan.CustomHours.Settings != nil, make([]*businessHoursResourceModelCustomHoursSettings, 0), nil)
+	for _, item := range dto.customHours.settings {
+		customHoursSetting = append(customHoursSetting, &businessHoursResourceModelCustomHoursSettings{
+			Weekday: item.weekday,
+			Type:    item.typ,
+			From:    lo.Ternary(item.typ.ValueInt32() == 2, item.from, types.StringNull()),
+			To:      lo.Ternary(item.typ.ValueInt32() == 2, item.to, types.StringNull()),
+		})
+	}
+	customHours := &businessHoursResourceModelCustomHours{
+		Type:                dto.customHours.typ,
+		AllowMembersToReset: dto.customHours.allowMembersToReset,
+		Settings:            customHoursSetting,
+	}
+	callHandling := &businessHoursResourceModelCallHandling{
+		CallNotAnswerAction:        dto.callHandling.callNotAnswerAction,
+		ForwardToExtensionID:       dto.callHandling.forwardToExtensionID,
+		BusyOnAnotherCallAction:    dto.callHandling.busyOnAnotherCallAction,
+		BusyForwardToExtensionID:   dto.callHandling.busyForwardToExtensionID,
+		AllowCallersCheckVoicemail: dto.callHandling.allowCallersCheckVoicemail,
+		AllowMembersToReset:        dto.callHandling.allowMembersToReset,
+		AudioWhileConnectingID:     dto.callHandling.audioWhileConnectingID,
+		CallDistribution: lo.TernaryF(plan.CallHandling != nil && plan.CallHandling.CallDistribution != nil && dto.callHandling.callDistribution != nil, func() *businessHoursResourceModelCallHandlingCallDistribution {
+			return &businessHoursResourceModelCallHandlingCallDistribution{
+				HandleMultipleCalls:          dto.callHandling.callDistribution.handleMultipleCalls,
+				RingDuration:                 dto.callHandling.callDistribution.ringDuration,
+				RingMode:                     dto.callHandling.callDistribution.ringMode,
+				SkipOfflineDevicePhoneNumber: dto.callHandling.callDistribution.skipOfflineDevicePhoneNumber,
+			}
+		}, func() *businessHoursResourceModelCallHandlingCallDistribution {
+			return nil
+		}),
```

Comment:
[nits]
You can replace elseFunc to `lo.NIl`.

Reply:
In my imagine, we can write like following. But it cannot be built. So I just keep it.

```go
lo.TernaryF(func () {
...
}, lo.Nil)
```

Reply:
Oh, I see.